src/main/java/net/sf/jabref/logic/importer/ParserResult.java

```diff
@@ -167,7 +163,19 @@ public void setErrorMessage(String errorMessage) {
     }
 
     public BibDatabaseContext getDatabaseContext() {
-        return new BibDatabaseContext(base, metaData, file);
+        if (Objects.isNull(this.bibDatabaseContext)) {
```

Comment:
I think the good old  `something == null` is more readable. According to the documentation, Objects.isNull just exists to be used as a predicate (i.e. ̀`filter(Objects::isNull)`)